### PR TITLE
Add ML and data science terms for issue #1032

### PR DESCRIPTION
**Pull request by Vlasovets**
- Added 52 new English term definitions covering:
  * Data structures (array, binary_tree, heap, linked_list, queue, set, stack)
  * Algorithms (backtracking, divide_and_conquer, dynamic_programming, greedy_algorithm, hashing, memoization, searching, sorting)
  * Data preparation (data_analysis, data_analytics, data_cleaning, data_preprocessing, dataset, missing_data)
  * Statistics and ML basics (bayesian_statistics, hypothesis_testing, probability)
  * ML concepts (activation_function, batch_size, label, loss_function, learning_rate)
  * ML models (knn, lstm, pca, recurrent_neural_network, regression, transformer)
  * Evaluation metrics (auc, f1_score, precision, recall)
  * ML workflow (model_evaluation, model_training, test_set, training_set, validation_set)
  * Other concepts (aliased, tensorflow, underfitting, variance_ml)

- All entries include proper cross-references to related terms
- Glossary validated and sorted alphabetically by slug
- Definitions written for data science learners

Addresses [#1032](https://github.com/carpentries/glosario/issues/1032)

### CODE REVIEW

**Comment by elletjies** [comment]
@all-contributors please add @Vlasovets for translation

**Comment by gvwilson** [comment]
Please break into separate pull requests for each new definition - 52 in one PR is too large to review.
